### PR TITLE
Add Deformable DETR model from mmdet

### PR DESCRIPTION
**Pull request by dnth**
This PR adds the Deformable DETR model into IceVision.

### CODE REVIEW

**Comment by FraPochetti** [comment]
Thanks for your contribution @dnth !
I have no clue of what Deformable DETR is though :)
Do you have a link to share?

**Comment by dnth** [comment]
> Thanks for your contribution @dnth ! I have no clue of what Deformable DETR is though :) Do you have a link to share?

Actually I dont have any clue too >.<

I just notice on the mmdet repo
https://github.com/open-mmlab/mmdetection/tree/master/configs/deformable_detr

**Comment by dnth** [comment]
From my experiment, i notice that it scores consistently better result compared to the DETR on the fridge dataset in the `getting_started_object_detection.ipynb` notebook

**Comment by FraPochetti** [comment]
I didn't look into the PR in detail yet, but I can already share a piece of feedback.
1. Can you please remove this cell output from the notebook?
image

2. Can you also re-train with VFNet? The `getting_started_object_detection.ipynb` is basically the entry point of our documentation and it just looks better to have a model scoring 95%+ rather than 80%.

File: icevision/models/mmdet/utils.py [hunk 1]
```diff
@@ -47,7 +47,7 @@ def param_groups(model):
         layers += [getattr(body, l) for l in body.res_layers]
 
     # add the neck module if it exists (DETR doesn't have a neck module)
-    layers += [model.neck for name in model.named_modules() if name == "neck"]
+    layers += [module for name, module in model.named_modules() if name == "neck"]
```
Comment (FraPochetti) [inline icevision/models/mmdet/utils.py:50]: This change was change merged in this [PR](https://github.com/airctic/icevision/pull/1029)

Comment (FraPochetti) [reply icevision/models/mmdet/utils.py:50]: This is the line which is causing the conflict. Are you able to resolve it?

**Comment by FraPochetti** [comment]
Thanks a ton!

**Pull request merged**